ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx

```diff
@@ -72,6 +75,17 @@ export const BeamEditor: React.FC = () => {
 
   const handleNsfwCheckbox = () => {
     setIsNsfw(!isNsfw);
+
+    if (!isNsfw) {
+      range(0, blocksInUse.length).forEach(key => {
+        setNsfwBlocks(new Map(nsfwBlocks.set(key, true)));
```

Comment:
Can you please explain the logic in this loop?

Reply:
It sets all blocks on the editor as nsfw blocks when the nsfw checkbox on the header is set. 
![image](https://github.com/AKASHAorg/akasha-core/assets/35398733/a61ae483-90a9-4c86-b1e3-7f98bb681a6f)

Reply:
Why use `range` function from lodash instead of a normal for loop?

Reply:
also is there a need to create n*Map objects where `n` is the number of blocks?

Reply:
There are also fundamental differences between a forEach function and a for loop. You can never exit from a forEach.

Reply:
Indeed

Reply:
also the overhead of that range function from lodash is crazy https://github.com/lodash/lodash/blob/main/src/.internal/baseRange.ts

Reply:
It's not worth it. Initially I considered the overhead as well but weighed over readability.

Reply:
> also is there a need to create n*Map objects where n is the number of blocks? can you please explain the logic?

React can't re-render if I modify existing map. That's why I am doing this :

`setNsfwBlocks(new Map(nsfwBlocks.set(key, true)))`

Reply:
ok but couldn't this be done in one update using just 1 Map object?

Reply:
In this case you are right. We can do that since all have the same value for the key otherwise I would have needed to create new Map n times.

Reply:
Thanks!